cmd/sonobuoy/app/master.go

```diff
@@ -61,14 +60,12 @@ func runMaster(cmd *cobra.Command, args []string) {
 	}
 
 	// Run Discovery (gather API data, run plugins)
-	if errcount := discovery.Run(kubeClient, cfg); errcount > 0 {
-		exit = 1
-	}
+	errcount := discovery.Run(kubeClient, cfg)
 
 	if noExit {
 		logrus.Info("no-exit was specified, sonobuoy is now blocking")
 		select {}
 	}
 
-	os.Exit(exit)
+	os.Exit(errcount)
```

Comment:
I don't love this? I think exit code should be 1 no matter what, unless we want to assign extra meanings for the exit codes